keps/sig-network/1669-graceful-termination-local-external-traffic-policy/README.md

```diff
@@ -125,8 +132,9 @@ All existing E2E tests for Services should continue to pass.
 
 #### Alpha
 
-* kube-proxy internally tracks the terminating condition of an endpoint.
-* feature is only enabled if the feature gate `EndpointSliceTerminatingCondition` is on.
+* kube-proxy internally tracks the `terminating` and `serving` condition from EndpointSlice
```

Comment:
L127 - are we really proposing a test that is "flaky by design"?

Reply:
Not sure what you mean by this, can you expand?

Reply:
In line 127 you write:

"This test may be marked "Flaky" as the behavior is largely also dependant on the cloud provider's loadbalancer."

I'm really against that - we should create a test that won't be flaky by design.

[We have many tests that depend on cloud-provider that aren't flaky.]

Reply:
> We have many tests that depend on cloud-provider that aren't flaky.

no flakes, we'll figure it out, I plan to work on this with Andrew too

Reply:
I think the reality is that the behavior is going to depend on the cloud provider tests though... so if they are flaky then testing this behavior is also going to be flaky. Not saying we shouldn't fix this, but just reflecting the reality of the situation.

Reply:
we can do it, GCE loadbalancers test have large timeouts, I think that we can keep an endpoint "terminating" for 5 mins if necessary ... I'm also working on some mocking for the loadbalancer that can test the functionality in kind, but this is a more long term solution

Reply:
> think that we can keep an endpoint "terminating" for 5 mins if necessary

Yes, but you can't control the time between the health check that will remove nodes from a node pool, so the behavior will be inherently difficult to test. 

I think I agree with both of you, let's settle for flaky tests, my original intent was just stating that flakiness of these tests are gonna be a real issue we have to tackel.

Reply:
> think that we can keep an endpoint "terminating" for 5 mins if necessary

Yes, but you can't control the time between the health check that will remove nodes from a node pool, so the behavior will be inherently difficult to test. 

I think I agree with both of you, let's not settle for flaky tests, my original intent was just stating that flakiness of these tests are gonna be a real issue we have to tackel.

Reply:
@andrewsykim it's definitely a risk for the KEP; in my experience these tests are hard to get right, especially with cloud LBs we don't control in any way, so we shouldn't underestimate the amount of time it will take to de-flake the tests we write for this feature. I hope I'm wrong.

Reply:
Don't disagree with anything there, just noting the risk

Reply:
I removed the comment about flaky tests to clear things up